conans/client/tools/net.py

```diff
@@ -60,21 +86,36 @@ def download(url, filename, verify=True, out=None, retry=None, retry_wait=None,
     retry_wait = retry_wait if retry_wait is not None else config.retry_wait
     retry_wait = retry_wait if retry_wait is not None else 5
 
-    downloader = FileDownloader(requester=requester, output=out, verify=verify, config=config)
+    url = [url] if not isinstance(url, list) else url
     checksum = sha256 or sha1 or md5
-    # The download cache is only used if a checksum is provided, otherwise, a normal download
+
+    downloader = FileDownloader(requester=requester, output=out, verify=verify, config=config)
     if config and config.download_cache and checksum:
         downloader = CachedFileDownloader(config.download_cache, downloader, user_download=True)
-        downloader.download(url, filename, retry=retry, retry_wait=retry_wait, overwrite=overwrite,
-                            auth=auth, headers=headers, md5=md5, sha1=sha1, sha256=sha256)
-    else:
-        downloader.download(url, filename, retry=retry, retry_wait=retry_wait, overwrite=overwrite,
-                            auth=auth, headers=headers)
-        if md5:
-            check_md5(filename, md5)
-        if sha1:
-            check_sha1(filename, sha1)
-        if sha256:
-            check_sha256(filename, sha256)
 
-    out.writeln("")
+    def _download_file(downloader, url):
+        # The download cache is only used if a checksum is provided, otherwise, a normal download
+        if isinstance(downloader, CachedFileDownloader):
+            downloader.download(url, filename, retry=retry, retry_wait=retry_wait,
+                                overwrite=overwrite, auth=auth, headers=headers, md5=md5,
+                                sha1=sha1, sha256=sha256)
+        else:
+            downloader.download(url, filename, retry=retry, retry_wait=retry_wait,
+                                overwrite=overwrite, auth=auth, headers=headers)
+            if md5:
+                check_md5(filename, md5)
+            if sha1:
+                check_sha1(filename, sha1)
+            if sha256:
+                check_sha256(filename, sha256)
+        out.writeln("")
+
+    for index, url_it in enumerate(url):
+        try:
+            _download_file(downloader, url_it)
+            break
+        except (ConanConnectionError, NotFoundException, ConanException) as error:
```

Comment:
All of them inherit from `ConanException`

```suggestion
        except ConanException as error:
```